exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggTemplate.java

```diff
@@ -325,10 +325,13 @@ public AggOutcome doWork() {
                 if (EXTRA_DEBUG_1) {
                   logger.debug("Received new schema.  Batch has {} records.", incoming.getRecordCount());
                 }
-                newSchema = true;
-                this.cleanup();
-                // TODO: new schema case needs to be handled appropriately
-                return AggOutcome.UPDATE_AGGREGATOR;
+                final BatchSchema newIncomingSchema = incoming.getSchema();
+                if ((! newIncomingSchema.equals(schema)) && schema != null) {
```

Comment:
Yeah, I thought this change might not be right.

So what do I do (to at least avoid the schema-change exception that I was getting from one of the test suites, but do so without introducing the bug you point out)?